lobby.html

```diff
@@ -533,7 +559,154 @@
+                    pixelsToCheck.push(x, y + 1);
+                    pixelsToCheck.push(x, y - 1);
+                }
+            }
+
+            // put the data back
+            ctx.putImageData(imageData, 0, 0);
+        }
+    }
+
+    function drawLine(context, x1, y1, x2, y2, color, lineWidth) {
+        // the coordinates must be whole numbers to improve performance.
+        // also, decimals as coordinates is not making sense.
+        x1 = Math.floor(x1);
+        y1 = Math.floor(y1);
+        x2 = Math.floor(x2);
+        y2 = Math.floor(y2);
+        lineWidth = Math.ceil(lineWidth);
+
+        color = hexToRgb(color);
+        color[3] = 255; //alpha channel
+
+        circleMap = generateCircleMap(Math.floor(lineWidth / 2));
+
+        offset = Math.floor(circleMap.length / 2);
+        const imageData = context.getImageData(0, 0, context.canvas.width, context.canvas.height);
+
+        for (ix = 0; ix < circleMap.length; ix++) {
+            for (iy = 0; iy < circleMap[ix].length; iy++) {
+                if (circleMap[ix][iy] === 1 || (x1 === x2 && y1 === y2 && circleMap[ix][iy] === 2)) {
+                    var newX1 = x1 + ix - offset;
+                    var newY1 = y1 + iy - offset;
+                    var newX2 = x2 + ix - offset;
+                    var newY2 = y2 + iy - offset;
+                    drawBresenhamLine(imageData, newX1, newY1, newX2, newY2, color);
+                }
+            }
+        }
+        context.putImageData(imageData, 0, 0);
+    }
+
+    function drawBresenhamLine(imageData, x0, y0, x1, y1, color) {
+        var dx = Math.abs(x1 - x0);
+        var dy = Math.abs(y1 - y0);
+        var sx = (x0 < x1) ? 1 : -1;
+        var sy = (y0 < y1) ? 1 : -1;
+        var err = dx - dy;
+
+        while (true) {
+            setPixel(imageData, x0, y0, color); // Do what you need to for this
+
+            if ((x0 === x1) && (y0 === y1)) break;
+            var e2 = 2 * err;
+            if (e2 > -dy) {
+                err -= dy;
+                x0 += sx;
+            }
+            if (e2 < dx) {
+                err += dx;
+                y0 += sy;
+            }
+        }
+    }
+
+    function generateCircleMap(radius) {
+        //console.log("generateCircleData: radius:" + radius);
+        var circleData = [];
+
+        for (x = 0; x < 2 * radius; x++) {
+            circleData[x] = [];
+            for (y = 0; y < 2 * radius; y++) {
+                distanceToRadius = Math.sqrt(Math.pow(radius - x, 2) + Math.pow(radius - y, 2));
+                if (distanceToRadius > radius) {
+                    circleData[x][y] = 0;
+                } else if (distanceToRadius < radius - 2) {
+                    //optimize for performance: fill circle only when mouse was not moved
+                    circleData[x][y] = 2;
+                } else {
+                    circleData[x][y] = 1;
+                }
+            }
+        }
+
+        return circleData;
+    }
+
```

Comment:
Whitespace :o

Reply:
you meant that few lines after that closing parantheses right? :D